### PR TITLE
separate sdw master and slave bus type

### PR DESCRIPTION
**Pull request by bardliao**
Some comments from Vinod. 
These patches except the one for sof_sdw_rt711.c will be squashed before sending to upstream.

### CODE REVIEW

**Comment by plbossart** [comment]
I am not sure at all those changes are valid. Is there any example where the host and devices have a different bus_type? I mean, it means they are on different buses, no?

**Comment by plbossart** [comment]
> I am not sure at all those changes are valid. Is there any example where the host and devices have a different bus_type? I mean, it means they are on different buses, no?

if you look at the emails, it was what Jaroslav suggested but he was ok with having them on the same bus type.

**Comment by plbossart** [comment]
@bardliao there is a nice counter example, see GreyBus:

https://elixir.bootlin.com/linux/latest/source/drivers/greybus/core.c#L81

**Comment by bardliao** [comment]
> > I am not sure at all those changes are valid. Is there any example where the host and devices have a different bus_type? I mean, it means they are on different buses, no?
> 
> if you look at the emails, it was what Jaroslav suggested but he was ok with having them on the same bus type.

Yes, but Vinod suggested the same thing, too. It would be better to prepare the patches while I am waiting for his confirmation.

**Comment by plbossart** [comment]
> > > I am not sure at all those changes are valid. Is there any example where the host and devices have a different bus_type? I mean, it means they are on different buses, no?
> > 
> > 
> > if you look at the emails, it was what Jaroslav suggested but he was ok with having them on the same bus type.
> 
> Yes, but Vinod suggested the same thing, too. It would be better to prepare the patches while I am waiting for his confirmation.

sure, but you may have to revert the changes later if GregKH objects...

File: drivers/soundwire/master.c [hunk 1]
```diff
@@ -30,12 +30,12 @@ struct device_type sdw_master_type = {
...
  * initializations and power-management are required.
  */
-struct sdw_master_device
-*sdw_master_device_add(struct device *parent,
-		       struct fwnode_handle *fwnode,
-		       struct sdw_link_ops *link_ops,
-		       int link_id,
-		       void *pdata)
+struct sdw_master_device*
+sdw_master_device_add(struct device *parent,
```
Comment (lyakh) [inline drivers/soundwire/master.c:34]: just put them on one line, please, makes 72 characters.

**Comment by lyakh** [comment]
> I am not sure at all those changes are valid. Is there any example where the host and devices have a different bus_type? I mean, it means they are on different buses, no?

@plbossart not sure I understand the sdw topology right, but it seems that sdw slaves have an sdw master as their parent device, right? That's certainly valid for parent and children to be on different busses, USB controllers can themselves be PCI devices etc.

**Comment by plbossart** [comment]
@bardliao take a look at /sys/bus/, it's rather confusing if you have 2 bus types...

````
root@plb:/sys/bus/soundwire-master/devices/sdw-master-0# ls
driver  firmware_node  power  sdw:0:25d:700:0  sdw:0:25d:711:0  subsystem  uevent

root@plb:/sys/bus/soundwire/devices# ls
sdw:0:25d:700:0  sdw:1:25d:1308:0  sdw:2:25d:700:0  sdw:3:25d:700:0
sdw:0:25d:711:0  sdw:1:25d:700:0   sdw:2:25d:701:0  sdw:3:25d:715:0
````

so the same devices appear on one bus and as children of devices on another bus.
Is this really necessary?

Compare with I2C and SPI, no one else has a bus for the controllers and another for devices. It's not logical to me.

**Comment by plbossart** [comment]
> > I am not sure at all those changes are valid. Is there any example where the host and devices have a different bus_type? I mean, it means they are on different buses, no?
> 
> @plbossart not sure I understand the sdw topology right, but it seems that sdw slaves have an sdw master as their parent device, right? That's certainly valid for parent and children to be on different busses, USB controllers can themselves be PCI devices etc.

right, but no one defined a USB-controller bus type. It's USB.

````
root@plb:/sys/bus/usb/devices# ls
1-0:1.0   1-10:1.1  1-4:1.1  1-7      1-7:1.2  2-0:1.0  2-3:2.1  2-3:2.4  3-0:1.0  usb2
1-10      1-4       1-4:1.2  1-7:1.0  1-7:1.3  2-3      2-3:2.2  2-3:2.5  4-0:1.0  usb3
1-10:1.0  1-4:1.0   1-4:1.3  1-7:1.1  1-7:1.4  2-3:2.0  2-3:2.3  2-3:2.6  usb1     usb4
````

**Comment by bardliao** [comment]
> @bardliao take a look at /sys/bus/, it's rather confusing if you have 2 bus types...
> 
> ```
> root@plb:/sys/bus/soundwire-master/devices/sdw-master-0# ls
> driver  firmware_node  power  sdw:0:25d:700:0  sdw:0:25d:711:0  subsystem  uevent
> 
> root@plb:/sys/bus/soundwire/devices# ls
> sdw:0:25d:700:0  sdw:1:25d:1308:0  sdw:2:25d:700:0  sdw:3:25d:700:0
> sdw:0:25d:711:0  sdw:1:25d:700:0   sdw:2:25d:701:0  sdw:3:25d:715:0
> ```
> 
> so the same devices appear on one bus and as children of devices on another bus.
> Is this really necessary?
> 
> Compare with I2C and SPI, no one else has a bus for the controllers and another for devices. It's not logical to me.

Ok. This convince me. :)

**Comment by plbossart** [comment]
@bardliao looks like there's no longer a debate on this if I look at the latest comments on alsa-deve, is this PR still relevant or can it be closed?

**Comment by bardliao** [comment]
> @bardliao looks like there's no longer a debate on this if I look at the latest comments on alsa-deve, is this PR still relevant or can it be closed?

Yeah, let's close it now

**Pull request closed without merging**